design-proposals/eim-standalone-managed-en.md

```diff
@@ -0,0 +1,236 @@
+4. A user is prompted for inputs. The user will be asked for the orchestrator FQDN, proxy settings (the already set proxy settings should be presented to the user),
+   and SEN onboarding user credentials. The CLI tool should validate that all required parameters are set and the orchestrator should be reachable at this point.
+   Moreover, the CLI tool can ask for additional input from the user such as which Local Accounts, Site or metadata to configure for a given EN (user must select a configuration that
+   already exists on the orchestrator and OM should be responsible for validating if that configuration exists in the Inventory).
+   Also, the CLI tool retrieves hardware info (UUID, Serial Number), OS info (OS version and distro from `/etc/os-release`), current the Secure Boot and Full-Disk Encryption settings,
+   MAC/IP address of the management interface.
+5. The CLI tool downloads the CA certificate from the Edge Orchestrator.
+   Note that it may be less secure, but more automated to download it via CLI. We can revisit this step and require
+   users to provide certificate as a user input. Also, this step is only required when the orchestrator uses self-signed certificate.
+6. The CA certificate is downloaded and saved to the local filesystem for the future use.
+7. The CLI tool communicates with the southbound Keycloak endpoint to retrieve JWT token based on SEN onboarding user credentials.
+8. The JWT token is provided to the CLI tool and used in subsequent calls to the Onboarding Manager's gRPC endpoint.
+9. At this point, the CLI tool gathered all user inputs (and validated them) and HW/OS info. It invokes the
+    Onboarding Manager's gRPC endpoint to trigger the onboarding process.
+    Note that the CLI tool should retry if any step above fails on the orchestrator side and the Onboarding Manager should be able to
+    handle partial states.
+10. Onboarding Manager validates the JWT role as it currently does for all gRPC APIs.
+11. Onboarding Manager reads the OS info and queries OS profile from Inventory based on the OS version and OS distro.
+    OS version should uniqely identify the OS profile. Note that this is true for EMT OS profiles, but may not be true for mutable OSes.
+    See open issues for more considerations on the support of mutable OSes.
+12. The Onboarding Manager creates a dedicated cloud-init configuration for SEN. The current cloud-init library can be used.
+    Note that the SEN will only require a subset of current cloud-init that is generated for remote provisioning.
+13. Once OS profile is found and matched, all required HW/OS info provided and cloud-init generated, the OM creates Host and Instance resources.
+    The statuses should be set to Onboarded and Provisioned. Host's desired and current state should be set to `ONBOARDED`.
+    The Instance desired state should be set to `RUNNING`, while the current state to `UNSPECIFIED` (until BM agents are up and running).
```

Comment:
are you sure about this ? I feel we should modify the hrm to touch only the modern status and avoid the modification of the current state. I feel the current state should be own only by the OM